CMakeLists.txt

```diff
@@ -68,23 +69,42 @@ else()
     message(STATUS "Disabled Component: gnssro")
 endif()
 
+if( eckit_FOUND AND oops_FOUND AND jedi_base_FOUND AND ioda_engines_FOUND AND bufr_FOUND )
+    set(iodaconv_bufr_ENABLED True)
+    message(STATUS "Found: eckit")
+    message(STATUS "Found: oops")
+    message(STATUS "Found: jedi_base")
+    message(STATUS "Found: ioda_engines")
+    message(STATUS "Found: bufr")
+    message(STATUS "Enabled Component: bufr")
+else()
+    set(iodaconv_bufr_ENABLED False)
+    if(NOT eckit_FOUND)
+        message(STATUS "NOT-Found: eckit - BUFR converter disabled")
+    endif()
+    if(NOT oops_FOUND)
+        message(STATUS "NOT-Found: oops - BUFR converter disabled")
+    endif()
+    if(NOT jedi_base_FOUND)
+        message(STATUS "NOT-Found: jedi_base - BUFR converter disabled")
+    endif()
+    if(NOT ioda_engines_FOUND)
+        message(STATUS "NOT-Found: ioda_engines - BUFR converter disabled")
+    endif()
+    if(NOT bufr_FOUND)
+        message(STATUS "NOT-Found: bufr - BUFR converter disabled")
+    endif()
+    message(STATUS "Disabled Component: bufr")
+endif()
+
+
+include( iodaconv_compiler_flags )
+include_directories( ${IODACONV_INCLUDE_DIRS} ${IODACONV_EXTRA_INCLUDE_DIRS} )
```

Comment:
We are moving away from using this method of setting compiler flags, which is a holdover from ecbuild.  If you require a specific option use `target_compile_options()` or `target_compile_features()`, and if necessary use a conditional against `CMAKE_<LANG>_COMPILER_ID` and `CMAKE_<LANG>_COMPILER_Version`.  See for example how we do this for the GNSSRO converter regarding gfortran-10 specific flags.  If there are system specific flags required they should go into the system-specific toolchain under `jedicmake`.

Reply:
Also there is no need for `include_directories()` commands.  Those should be handled by the [`INTERFACE_INCLUDE_DIRECTORIES` ]( `https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html`) target property of each lineked target dependency.

Reply:
I don't agree with removing `iodaconv_compiler_flags`.
Compiler flags pertaining to a repository are stored there. oops does this, soca, fv3jedi, ufo all do this.
Compiler flags pertaining to a platform/system are handled at the toolchain level e.g. AVX2 instruction flags, SIMD flags etc.

If an executable/library needs specific flags, they can be added at the level of the source code and not in the location of iodaconv_compiler_flags.
For e.g. debug, release, bit flags for the project should reside in iodaconv_compiler_flags.
executable specific flags. e.g. -fconvert=big-endian when dealing with IO of binary files belong with the application.

The gfortran-10 flags in gnssro should be in `iodaconv_compiler_flags` to avoid repetition of this code everywhere a fortran source code is compiled.

Reply:
I am fine with removing `include_directories`

Reply:
Actually we've been discussing this in the core JEDI meetings for some time, and it is a current goal to remove this way of setting CFLAGS.  Dan has already removed them from [fv3-jedi/cmake](https://github.com/JCSDA/fv3-jedi/tree/develop/cmake) and the rest of the repositories will eventually move that way too.  Also all the ecmwf packages have already removed them.  They preventing the use of the built-in CFLAGS handling of ecbuild and CMake.  

The documentation for how ecbuild is designed to use CFLAGS in the comments of [ecbuild_compiler_flags.cmake](https://github.com/JCSDA/ecbuild/blob/release-stable/cmake/ecbuild_compiler_flags.cmake).

>  The procedure is as follows:
> 
>  1.  ecBuild does **not** set ``CMAKE_<lang>_FLAGS`` i.e. the user can set
>      these via ``-D`` or the CMake cache and these will be the "base" flags.
> 
>  2.  ecBuild **overwrites** ``CMAKE_<lang>_FLAGS_<btype>`` in the CMake cache
>      for all build types with compiler specific defaults for the currently
>      loaded compiler i.e. any value set by the user via ``-D`` or the CMake
>      cache **has no effect**.
> 
>  3.  Any value the user provides via ``ECBUILD_<lang>_FLAGS`` or
>      ``ECBUILD_<lang>_FLAGS_<btype>`` **overrides** the corresponding
>      ``CMAKE_<lang>_FLAGS`` or ``CMAKE_<lang>_FLAGS_<btype>`` **without being
>      written to the CMake cache**.

We want to move to using toolchains for specifying the jedi-system-wide specialized debug flags.  The problem is that changing a single debug flag requires modifying every single repository with the old way of setting flags, and because it doesn't follow the ecbuild mechanisms for setting flags, we cannot override these compiler flags easily with a toolchain.
 
When there are target specific flags that must only be applied for debug builds, they should be added with a generator expression  `target_compile_options(tgt PRIVATE $<$<CONFIG:Debug>:-my-special-debug-flag>)`.  Setting flags like this allows the distinction between PRIVATE, PUBLIC, and INTERFACE inheritance of imported target flags.

Reply:
@markjolah 

I agree that target specific flags should be applied on the target itself.  However, where do you set general repo flags such as `-g -traceback`? Are these flags at toolchain level? If so, where is the toolchain repo, and I can add that to the repo.

There is no use of `ecbuild_compiler_flags` macro here. It is a simple include of `iodaconv_compiler_flags` that sets the compiler ID specific flags and for this repo alone.

I am following logic from the OOPS repository. OOPS still has `oops_compiler_flags`.
Where do you discuss these items? I don't think I heard anything to this tune in the JEDI weekly round table discussions.